src/fileConverter.ts

```diff
@@ -77,77 +114,89 @@ export class FileConverter {
+        // We need to open an issue on this as the @types/sarif version and the multi-tool version can stray apart because
+        // the multi-tool code package has no exports to say what version of the schema it is on.
+        if (parsedVersion.original === mTCurVersion.original) {
+            return {
+                upgradedNeeded: 'No',
+                parsedVersion
+            };
         }
 
-        if (schema === 'http://json.schemastore.org/sarif-2.1.0-rtm.1') {
-            // By passes a bug in the multitool, remove after fix https://github.com/microsoft/sarif-sdk/issues/1584
-            return false;
+        if (!sarifLog.$schema) {
+            return {
+                upgradedNeeded: 'Schema Undefined',
+                parsedVersion
+            };
         }
 
-        let parsedSchemaVer: SarifVersion | undefined;
-        let tryToUpgrade: boolean = false;
-        if (schema && (schema.startsWith('http://json.schemastore.org/sarif-') ||
-            schema.startsWith('https://schemastore.azurewebsites.net/schemas/json/sarif-'))) {
-            parsedSchemaVer = FileConverter.parseSchema(schema);
-
-            if (!parsedSchemaVer) {
-                return false;
-            }
-
-            const mTCurSchemaVersion: SarifVersion = FileConverter.MultiToolCurrentSchemaVersion;
-
-            tryToUpgrade = (parsedSchemaVer.original !== mTCurSchemaVersion.original) &&
-                FileConverter.isOlderThenVersion(parsedSchemaVer, FileConverter.MultiToolCurrentSchemaVersion);
-        } else {
-            return false;
+        // This is parsing the "$schema" property from the SARIF json.
+        // I.e.
+        // {
+        //   "$schema": "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.5.json",
+        //
+        const parsedSchemaVersion: SarifVersion | undefined = sarifLog.$schema !== undefined ? FileConverter.parseSchema(sarifLog.$schema) : undefined;
+        if (!parsedSchemaVersion) {
+            return {
+                upgradedNeeded: 'Could Not Parse Schema',
+                parsedVersion
+            };
         }
 
-        if (tryToUpgrade) {
-            // The upgrade process does not need to block the return of this function
-            // as it open the converted document when it is done whic starts
-            // the read SARIF cycle again.
-            // tslint:disable-next-line: no-floating-promises
-            FileConverter.upgradeSarif(doc, parsedVer, parsedSchemaVer);
-        } else {
-                await vscode.window.showErrorMessage(localize(
-                    'converterTool.UpgraderErrorMessage',
-                    "Sarif version '{0}'(schema '{1}') is not yet supported by the Viewer. Make sure you have the latest extension version and check https://github.com/Microsoft/sarif-vscode-extension for future support.",
-                    version, schema));
+        if (!sarifLog.$schema.startsWith('http://json.schemastore.org/sarif-') &&
```

Comment:
>http: [](start = 42, length = 5)

The protocol/scheme is case insensitive I do believe.

Reply:
You are correct. Fixing this check.

---
In reply to: [408274949](https://github.com/microsoft/sarif-vscode-extension/pull/241#discussion_r408274949) [](ancestors = 408274949)